de.metas.handlingunits.client/src/main/java/de/metas/handlingunits/client/terminal/editor/model/impl/MovementsAnyWarehouseModel.java

```diff
@@ -87,4 +85,40 @@ protected void createMovements()
 
 	}
 
+	/**
+	 * Get the warehouses existing for the hus' organization , excluding those which currently contain the HUs given as parameters
+	 * 
+	 * @param hus
+	 * @return
+	 */
+	public static List<org.compiere.model.I_M_Warehouse> getWarehousesUnlessOfHUs(final List<I_M_HU> hus)
+	{
+		if (hus.isEmpty())
+		{
+			// should never happen
+			return Collections.emptyList();
+		}
+
+		// used for deciding the org and context
+		final I_M_HU firstHU = hus.get(0);
+
+		final int orgId = firstHU.getAD_Org_ID();
+		final Properties ctx = InterfaceWrapperHelper.getCtx(firstHU);
+
+		final List<org.compiere.model.I_M_Warehouse> warehouses = Services.get(IWarehouseDAO.class).retrieveForOrg(ctx, orgId);
+
+		final ImmutableList<org.compiere.model.I_M_Warehouse> huWarehouses = hus
```

Comment:
we suggest to instead do
```java
final List<org.compiere.model.I_M_Warehouse> huWarehouses = hus
                 .stream()
                 .map(I_M_HU::getM_Locator_ID)
                 .distinct()
                 .map(id -> InterfaceWrapperHelper.load(id, I_M_Locator.class))
                 .map(org.compiere.model.I_M_Locator::getM_Warehouse_ID)
                 .distinct()
                 .map(id -> InterfaceWrapperHelper.load(id, I_M_Warehouse.class))
                 .collect(ImmutableList.toImmutableList());
```

Btw, do you know if `distinct()` depends on the streamed items to be ordered?

Reply:
From the examples I found online, (e.g. https://howtodoinjava.com/java-8/java-stream-distinct-examples/)  the stream doesn't have to be ordered.